app/src/renderer/vuex/modules/blockchain.js

```diff
@@ -61,6 +79,60 @@ export default ({ commit, node }) => {
         })
       })
     },
+    async queryTxInfo({ state, dispatch }, height) {
+      let blockTxInfo = state.blockTxs.find(
+        b => b.length && b[0].height === height
+      )
+      if (blockTxInfo) {
+        return blockTxInfo
+      }
+      try {
+        blockTxInfo = await dispatch("getTxs", {
+          key: 0,
+          len:
+            state.block && state.block.data && state.block.data.txs
+              ? state.block.data.txs.length
+              : 0,
+          txs:
+            state.block && state.block.data && state.block.data.txs
+              ? state.block.data.txs.slice(0)
+              : []
+        })
+        blockTxInfo && state.blockTxs.push(blockTxInfo)
+        return blockTxInfo
+      } catch (error) {
+        return Promise.reject(error)
+      }
+    },
+    getTxs({ state, commit, dispatch }, { key, len, txs }) {
+      //  this function is recursice promie used as an async loop in order to query all tx
+      // found in a block. it's made similarly to queryBlockInfo only there
+      // is more than one async call to make. the txstring is included but might not
+      // actually be useful. etherscan.io includes something similar but it's seldom helpful
+      return new Promise(async (resolve, reject) => {
+        if (key >= len) return resolve(txs)
+        let txstring = atob(txs[key])
+        let hash = await convertTx(txs[key])
+        node
+          .txs(hash)
```

Comment:
Why mix `await` and `.then` syntax?

Reply:
refactored all the promises to be more concise